### PR TITLE
Add german Bible book names

### PR DESCRIPTION
**Pull request by blankton2**
Add a de.json file that contains the german names of the bible books

### CODE REVIEW

**Review by tim-hub — CHANGES_REQUESTED** [review]
Thanks for the contribution

The test failed due to finding some duplication of names, I tried to find which book it is, but have not found it yet.

https://github.com/Antioch-Tech/bible-reference-toolkit/actions/runs/7420564775/job/20250973309?pr=14

File: packages/bible-book-names-intl/src/data/translations/de.json [hunk 1]
```diff
@@ -0,0 +1,548 @@
...
+    ],
+    "startNumber": 3
+  },
+  "4": {
+    "name": "Mose",
+    "shortNames": [
+      "Nu",
+      "Num",
+      "Mo"
+    ],
```
Comment (tim-hub) [inline packages/bible-book-names-intl/src/data/translations/de.json:36]: I see, the duplication is from the first 5 books from Mose,

Comment (tim-hub) [reply packages/bible-book-names-intl/src/data/translations/de.json:36]: I am not sure in German, does it have a book name like `Genesis` for `1 Mose`

Comment (Tecur) [reply packages/bible-book-names-intl/src/data/translations/de.json:36]: Yes, `Genisis` is also possible, but less common. `1. Mose` would be the preferred book name. 

If  `startNumber` is not working for the 5 books from Mose, we could call it `Genesis` in the `name` field and `1. Mose` in `shortNames`.

Comment (blankton2) [reply packages/bible-book-names-intl/src/data/translations/de.json:36]: > I am not sure in German, does it have a book name like `Genesis` for `1 Mose`

The books from Moses are also called Genesis, Exodus, ... 
But, like stated by Tecur, it is not common to reference them in this way (at least in evangelical churches).
Genesis1:1 is written 1.Mose1,1
A Paragraph: 2.Mose5,1-4

Comment (tim-hub) [reply packages/bible-book-names-intl/src/data/translations/de.json:36]: @Tecur 

> Yes, `Genisis` is also possible, but less common. `1. Mose` would be the preferred book name.
> 
> If `startNumber` is not working for the 5 books from Mose, we could call it `Genesis` in the `name` field and `1. Mose` in `shortNames`.



Do you mind move it it to shortNames? :)

Comment (Tecur) [reply packages/bible-book-names-intl/src/data/translations/de.json:36]: Sure. What is the correct syntax in this case? `"1. Mose"`, `"1.Mose"` or `"1Mose"`?

Comment (Tecur) [reply packages/bible-book-names-intl/src/data/translations/de.json:36]: @blankton2 Could you give me write access to your fork so I can add these changes directly to this PR?

I would also add some more abbreviations and name variants (according to the [german Wikipedia article](https://de.wikipedia.org/wiki/Liste_biblischer_B%C3%BCcher))
e.g. Zefanja <-> Zephanja

File: packages/bible-book-names-intl/src/data/translations/de.json [hunk 2]
```diff
@@ -0,0 +1,578 @@
...
+      "Tt",
+      "Ts"
+    ],
+    "startNumber": 0
+  },
+  "57": {
+    "name": "Philemon",
+    "shortNames": [
+      "Pm",
+      "Phil",
```
Comment (tim-hub) [inline packages/bible-book-names-intl/src/data/translations/de.json:492]:
```
src/tests/translations.test.ts:
  ✓ test translation structure > test english [2.27ms]
  ✓ test all translations > there are at least one valid JSON file [0.03ms]
  ✓ test all translations > test each translation [1.23ms]
  Phil is duplicated among other books
  
  Error: Expected: false
  Received: true
  ```
  
  I think Line 433 and Line 492, the short names are the same. Which cause the testing failing

**Review by tim-hub — CHANGES_REQUESTED** [review]

File: packages/bible-book-names-intl/src/data/translations/de.json (hunk 2 above)
Comment (Tecur) [reply packages/bible-book-names-intl/src/data/translations/de.json]: You're correct. Fixed it.

**Pull request merged**